.github/workflows/build.yml

```diff
@@ -0,0 +1,100 @@
+---
+name: Build RPM packages
+
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build_packages:
+    strategy:
+      matrix:
+        build_type:
+          - auto
+          - manual
+        version:
+          - 8
+          - 7
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup Python environment
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.8"
+          architecture: "x64"
+
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+
+      - name: Build packages
+        run: make ${{ matrix.build_type }}${{ matrix.version }}
+
+      - name: Cache built packages
+        uses: actions/cache@v2
+        id: restore-build
+        with:
+          path: _dist${{ matrix.version }}/*
+          key: ${{ github.sha }}-${{ matrix.build_type }}${{ matrix.version }}
+
+  publish_packages:
+    needs: build_packages
+    if: github.ref == 'refs/heads/master'
+    strategy:
+      matrix:
+        version:
+          - 8
+          - 7
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Setup Ruby environment
+        uses: ruby/setup-python@v1
+        with:
+          ruby-version: "3.0"
+
+      - name: Install PackageCloud CLI
+        run: gem install package_cloud
+
+      # following tasks will require adjustment:
+      - if: env.RPM_GPG_KEY_SECRET != ''
+        run: echo openssl aes-256-cbc -k ${RPM_GPG_KEY_SECRET} -in secret.asc.enc -out secret.asc -d
+
+      - if: env.RPM_GPG_KEY_PASSPHRASE != ''
+        run: echo ${RPM_GPG_KEY_PASSPHRASE} > .passphrase
+
+      - uses: actions/cache@v2
+        with:
+          path: _dist7/*
+          key: ${{ github.sha }}-auto7
+
+      - uses: actions/cache@v2
+        with:
+          path: _dist7/*
+          key: ${{ github.sha }}-manual7
+
+      - uses: actions/cache@v2
+        with:
+          path: _dist8/*
+          key: ${{ github.sha }}-auto8
+
+      - uses: actions/cache@v2
+        with:
+          path: _dist8/*
+          key: ${{ github.sha }}-manual8
+
+      - if: env.TRAVIS_REPO_SLUG == 'lest/prometheus-rpm'
```

Comment:
What do you think about changing it to check `secrets.PACKAGECLOUD_TOKEN` presence?

Reply:
Yes, and I will adjust it for whole job as well, not only for this step. It doesn't make sense to run other steps from this job, if it is not possible to publish packages.

Reply:
Awesome, thanks a lot!